lang/syn/src/codegen/program/handlers.rs

```diff
@@ -16,7 +16,7 @@ pub fn generate(program: &Program) -> proc_macro2::TokenStream {
             // on chain.
             #[inline(never)]
             #[cfg(not(feature = "no-idl"))]
-            pub fn __idl_dispatch(program_id: &Pubkey, accounts: &[AccountInfo], idl_ix_data: &[u8]) -> ProgramResult {
+            pub fn __idl_dispatch(program_id: &Pubkey, accounts: &[AccountInfo], idl_ix_data: &[u8]) -> ::anchor_lang::DynAnchorResult {
```

Comment:
These return type changes are where I'm not sure how to do the enum/no-dynamic dispatch approach—what should these return types be? It would depend on whether or not you've generated your own `#[error]` enum, right? I definitely agree that it's possible to have a nice static return type here (only a static number of error enums in your program), but I wasn't sure how to actually do it.